go/data-source-sql/countries/countries.go

```diff
@@ -24,6 +24,12 @@ type Country struct {
 	Name string `json:"name"`
 }
 
+type CountryCodeUndefined string
+
+func (code CountryCodeUndefined) Error() string {
+	return fmt.Sprintf("No country defined for code ‘%v’", string(code))
```

Comment:
When you are creating a custom type which simply refers to a basic type (string, int, bool, etc.), you don't have to necessarily type cast it when passing it to `fmt.Sprintf` for example. You should be able to write the following instead (I think):

```go
return fmt.Sprintf("No country defined for code ‘%s’", code)
```

Reply:
I tried that first, that results in a call to the current function, infinite loop and stack overflow 😬